autopilot/contractor/hostfilter.go

```diff
@@ -165,7 +166,11 @@ func checkHost(gc gouging.Checker, sh scoredHost, minScore float64, period uint6
 	// calculate remaining host info fields
 	if !h.IsAnnounced() {
 		ub.NotAnnounced = true
-	} else if !h.Scanned {
+	} else if !h.Scanned ||
+		// NOTE: a v2 host might have been scanned before the v2 height so strictly
+		// speaking it is scanned but since it hasn't been scanned since, the
+		// settings aren't set so we treat it as not scanned
+		(h.IsV2() && h.V2Settings == (rhp4.HostSettings{})) {
```

Comment:
Did you consider moving it closer to the database? Have the store return unscanned hosts if this condition is true.

Reply:
Maybe. I'll try and see if it passes the tests. My first instinct was to not do it to avoid breaking the nice guarantee of `Scanned` not switching back from `false` to `true` but maybe it works just fine for the transition.